Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py

```diff
@@ -2774,6 +2775,109 @@ def long_running_loop():
+    Use-case:
+    This command is ideal for users encountering insufficient permissions errors when attempting to
+    execute an integration command.
+    By utilizing this command, the user can identify the current permissions associated with their token (app), compare them to
+    the required permissions for executing the desired command (detailed in the integration's docs), and determine any additional
+    permissions needed to be added to their application.
+    """
+    # Get the used token from the integration context:
+    access_token: str = get_graph_access_token()
+
+    # Decode the token and extract the roles:
+    if access_token:
+        decoded_token = jwt.decode(access_token, options={"verify_signature": False})
+
+        if AUTH_TYPE == CLIENT_CREDENTIALS_FLOW:
+            roles = decoded_token.get('roles', [])
+
+        else:  # Authorization code flow
+            roles = decoded_token.get('scp', '')
+            roles = roles.split()
+
+        if roles:
+            roles = sorted(roles)
+            hr = tableToMarkdown(f'The API permissions obtained for the used graph access token are: ({len(roles)})',
+                                 roles, headers=['Permission'])
+        else:
+            hr = 'No permissions obtained for the used graph access token.'
+
+    else:
+        hr = 'Graph access token is not set.'
+
+    demisto.debug(f"'microsoft-teams-token-permissions-list' command result is: {hr}")
+
+    result = CommandResults(
+        readable_output=hr
+    )
+
+    return_results(result)
+
+
+def create_messaging_endpoint_command():
+    """
+    Generates the messaging endpoint, based on the server url, the server version and the instance configurations.
+
+    The messaging endpoint should be added to the Demisto bot configuration in Microsoft Teams as part of the Prerequisites of
+    the integration's set-up.
+    Link to documentation: https://xsoar.pan.dev/docs/reference/integrations/microsoft-teams#create-the-demisto-bot-in-microsoft-teams
+    """
+    server_address = ''
+    messaging_endpoint = ''
+
+    # Get instance name and server url:
+    urls = demisto.demistoUrls()
+    instance_name = demisto.integrationInstance()
+    xsoar_url = urls.get('server', '')
+
+    # In case of an xsoar engine user - He must provide us with the engine address:
+    engine_url = demisto.args().get('engine_url', '')
+    if engine_url and not re.search(XSOAR_ENGINE_URL_REGEX, engine_url):
+        raise ValueError("Invalid engine URL - Please ensure that the engine_url includes the IP (or DNS name)"
+                         " and the port in use, and that it is in the correct format: `https://IP:port` or `http://IP:port`.")
+
+    if is_xsoar_on_prem():
+        if engine_url:  # user uses an engine
+            messaging_endpoint += urljoin(urljoin(engine_url, 'instance/execute'), instance_name)
+        else:
+            messaging_endpoint += urljoin(urljoin(xsoar_url, 'instance/execute'), instance_name)
```

Comment:
in case of using an engine, we dont have to add the "instance/execute/<instance_name>" suffix, 
AFAIK, the "instance/execute/<instance_name>" suffix is specific for cortex rerouting in long running integrations in general
see this article:  https://xsoar.pan.dev/docs/reference/articles/long-running-invoke